config/config_helpers.go

```diff
@@ -301,6 +307,93 @@ func getAWSAccountID(include *IncludeConfig, terragruntOptions *options.Terragru
 	return *identity.Account, nil
 }
 
+// Return the output from the state of another module, managed by terragrunt. This function will parse the provided
+// terragrunt config and extract the desired output from the remote state. Note that this will error if the targetted
+// module hasn't been applied yet. If you omit the output name, this will return a map of all the outputs from the
+// module.
+func getTerragruntOutput(params []string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (*cty.Value, error) {
+	// args check: make sure the expected args are passed
+	numParams := len(params)
+	if numParams == 0 {
+		return nil, errors.WithStackTrace(EmptyStringNotAllowed("the target terragrunt config parameter"))
+	}
+	if numParams > 2 {
+		return nil, errors.WithStackTrace(WrongNumberOfParams{Func: "get_output", Expected: "1 or 2", Actual: numParams})
+	}
+
+	// target config check: make sure the target config exists
+	targetConfig := params[0]
+	if util.IsDir(targetConfig) {
+		targetConfig = util.JoinPath(targetConfig, DefaultTerragruntConfigPath)
+	}
+	if !util.FileExists(targetConfig) {
+		return nil, errors.WithStackTrace(TerragruntOutputConfigNotFound{Path: targetConfig})
+	}
+
+	// use terragrunt running functions to extract the json output from the target config
+	var stdoutBuffer bytes.Buffer
+	stdoutBufferWriter := bufio.NewWriter(&stdoutBuffer)
+	targetOptions := terragruntOptions.Clone(targetConfig)
+	targetOptions.TerraformCliArgs = []string{"output", "-json"}
+	if numParams == 2 {
+		targetOptions.TerraformCliArgs = append(targetOptions.TerraformCliArgs, params[1])
+	}
+	targetOptions.Writer = stdoutBufferWriter
+	err := targetOptions.RunTerragrunt(targetOptions)
```

Comment:
Does this stream the `terragrunt output` call to stdout or stderr of the current session?

Reply:
It streams the `stderr` to the same `stderr` of the current session, since the `ErrWriter` is left untouched from the `Clone` function.

Reply:
We may want to test that calling `get_output` on dependencies does not dirty `stdout`. E.g., If I run `terragrunt output` on a module foo, that has a `terragrunt.hcl` with a `get_output` on module bar, we need to make sure that `stdout` contains _solely_ the result of aclling `terraform output` on foo, and not any of the logs from calling `terraform output` on bar.

Reply:
Verified in test that `stdout` isn't garbled by reading `terragrunt output` call to a module with `terragrunt_output` blocks defined on `stdout`.